internal/controller/authenticator.go

```diff
@@ -86,28 +85,31 @@ func (authn *Authenticator) DoesK8sSecretExist(ctx context.Context, secretName s
 	return err == nil
 }
 
-// WriteK8sSecret takes an OnboardResponse and writes it to a locally kept secret.
+// WriteK8sSecret updates the data in an existing Kubernetes Secret without overwriting annotations or labels.
 func (authn *Authenticator) WriteK8sSecret(ctx context.Context, keyData OnboardResponse, ClusterCreds string) error {
 	clientset, err := NewClientSet()
 	if err != nil {
 		authn.Logger.Error("Failed to create clientSet", zap.Error(err))
 		return err
 	}
 
-	secretData := map[string][]byte{
+	namespace := "illumio-cloud" // TODO: Make this configurable
```

Comment:
The Pod can discover its own namespace.
In the pod spec's `env`, add:
```
            - name: MY_POD_NAMESPACE
              valueFrom:
                fieldRef:
                  fieldPath: metadata.namespace
```
Per https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#use-pod-fields-as-values-for-environment-variables